WCAG 2.2 guidance updates (#3090)
alphagov/govuk-design-system: src/patterns/confirmation-pages/index.md

---

{% from "_example.njk" import example %}

> {% from "govuk/components/inset-text/macro.njk" import govukInsetText %}
> {% from "govuk/components/tag/macro.njk" import govukTag %}

Author: Remove all WCAG 2.2 messaging for Confirmation pages

```suggestion
```

Use this pattern to let users know they’ve completed a transaction.

{% set wcagCallout %}

{{ govukTag({
  text: "WCAG 2.2",
  classes: "app-tag"
}) }}

### New WCAG 2.2 criteria affects this pattern

To use ‘Confirmation pages' and meet the new Web Content Accessibility Guidelines (WCAG) 2.2 criteria, make sure that users can successfully:

> - [see and understand text in the green confirmation panel](/patterns/confirmation-pages/#wcag-see-understand-text)

Author: Remove Focus appearance message

```suggestion
```

> {% set wcagCallout %}
> 
> {{ govukTag({
>   text: "WCAG 2.2",
>   classes: "app-tag"
> }) }}
> 
> ### New WCAG 2.2 criteria affects this pattern
> 
> To use ‘Confirmation pages' and meet the new Web Content Accessibility Guidelines (WCAG) 2.2 criteria, make sure that users can successfully:
> 
> - [see and understand text in the green confirmation panel](/patterns/confirmation-pages/#wcag-see-understand-text)
> - [select and copy text within a confirmation page as well as its panel](/patterns/confirmation-pages/#wcag-select-copy-confirmation)
> 
> See the full list of [components and patterns affected by WCAG 2.2](/accessibility/WCAG-2.2/#components-and-patterns-affected-in-the-design-system).
> {% endset %}
> 
> {{ govukInsetText({
>   html: wcagCallout,
>   attributes: {
>     style: 'border-left-color: #1d70b8;'
>   }
> })}}
> 

Author: Remove all WCAG 2.2 messaging for Confirmation pages

```suggestion
```

- a link to your [feedback page](https://www.gov.uk/service-manual/service-assessments/get-feedback-page)
- a way for users to save a record of the transaction, for example, as a PDF

> <div class="app-wcag-22" id="wcag-see-understand-text" role="note">
>   {{ govukTag({
>     text: "WCAG 2.2",
>     classes: "app-tag"
>   }) }}
>   <p>Do not include links, buttons or other interactive elements in the green confirmation panel, as the focus state will not have a minimum contrast ratio of 3:1 with the panel’s background colour.</p>
> </div>
> 
> {{ example({group: "patterns", item: "confirmation-pages", example: "default", html: true, nunjucks: true, open: false, titleSuffix: "second", size: "xl"}) }}

Author: Taken out of WCAG 2.2 highlighting, kept message as standard guidance, saying that it's the Design System's focus style that does not meet minimum contrast.

```suggestion
Avoid including links, buttons or other interactive elements in the green confirmation panel, as the focus state will not meet the minimum contrast ratio of 3:1 with the panel’s background colour.
```

- starting a new application
- what to do or who to contact if they have a problem with their application

> <div class="app-wcag-22" id="wcag-select-copy-confirmation" role="note">
>   {{ govukTag({
>     text: "WCAG 2.2",
>     classes: "app-tag"
>   }) }}
>   <p>Allow users to select and copy text within a confirmation page as well as its <a href="/components/panel/">panel</a>. For example, users should be able to select and copy a booking number. This relates to WCAG 2.2 success criterion <a href="https://www.w3.org/WAI/WCAG22/Understanding/redundant-entry.html">3.3.7 Redundant Entry</a>.</p>
> </div>
> 
> Consider offering an option for the user to copy information in the panel component area into their device’s clipboard if there’s information that the user is expected to:
> 
> - save
> - remember
> - copy
> 

Author: Remove copy-paste message, there's no concern with either Redundant entry or Dragging movements

```suggestion
```
